accepted/2022/NuGetExeWarnWhenSDKProjectsGetPacked.md

```diff
@@ -1,99 +1,99 @@
+The proposal is to `error` whenever a SDK-based csproj pack is attempted, instructing the user to use `msbuild /t:pack` or `dotnet.exe pack` instead.
+
+## Motivation
+
+NuGet.exe does not support packing SDK-based, or PackageReference projects in general.
+If you run NuGet.exe pack on an SDK-based csproj, it might pack, but it will do it incorrectly.
+The pack SDK experience for PackageReference projects knows how to automatically manage the dependencies and makes multitargeted projects extremely easy to pack.
+It is recommended that SDK projects are managed by using `dotnet.exe` instead.
+
+## Explanation
+
+### Functional explanation
+
+#### How packing and SDK-based project with NuGet.exe behaves today
+
+Scenario 1: Single target framework SDK based projects, with .NET SDK installed on the machine.
+
+```console
+Attempting to build package from 'SDK.csproj'.
+MSBuild auto-detection: using msbuild version '17.4.0.37601' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
+Packing files from 'C:\Code\Temp\SDK\bin\Debug\net6.0'.
+WARNING: NU5115: Description was not specified. Using 'Description'.
+WARNING: NU5115: Author was not specified. Using 'Roki2'.
+WARNING: NU5128: Some target frameworks declared in the dependencies group of the nuspec and the lib/ref folder do not have exact matches in the other location. Consult the list of actions below:
+- Add a dependency group for net6.0 to the nuspec
+Successfully created package 'C:\Code\Temp\SDK\SDK.1.0.0.nupkg'.
+```
+
+The command packages the single assembly, but does not generate the correct dependencies or any other metadata in the same way that the pack SDK would.
+
+Scenario 2: Multiple target framework SDK based projects, with .NET SDK installed on the machine.
+
+```console
+Attempting to build package from 'SDK.csproj'.
+MSBuild auto-detection: using msbuild version '17.4.0.37601' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
+Error NU5012: Unable to find 'bin\Debug\SDK\bin\Debug\'. Make sure the project has been built.
+```
+
+NuGet.exe does not understand the concept of multitargetting and just merges the output paths.
+
+Scenario 3: SDK-based project, without the .NET SDK installed
+
+```console
+The SDK 'Microsoft.NET.Sdk' specified could not be found.  C:\Code\Temp\SDK\SDK.csproj
+```
+
+The parsing fails early.
+
+### Proposal
+
+When someone attempts to run pack on an SDK-based project, NuGet will fail with the following error:
+
+```console
+Error NU5049: The `pack` command for SDK-style projects is not supported, use `dotnet pack` to pack this project instead. You may set the ENABLE_LEGACY_NUGETEXE_CSPROJ_PACK environment variable to revert to the previous packing behavior.
+```
+
+This *breaking change* will be announced via blog well before the NuGet.exe carrying the change ships.
+
+### Technical explanation
```

Comment:
Will there be a new `SearchQueryService` resource version so that server can explicitly state they support owners in the response JSON?
https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource#versioning

Alternatively, I guess you are checking for the presence of the `owners` property?

We didn't do this for deprecation or vulnerabilities but depending on whether you want to make owners vs. authors a feed level decision or an individual search result level decision, it might help to have it declared in the service index. Something to consider.

Reply:
Added to Future Possibilities:
> Stop falling back to `author` when a package source supports Trusted Owners.